### PR TITLE
Disabled flaky SuggestPermissionIfWidevineDetected on x86 only

### PR DESCRIPTION
**Pull request by simonhong**
fix https://github.com/brave/brave-browser/issues/22386

Resolves 

## Submitter Checklist:

- [ ] I confirm that no security/privacy review [is needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or that I have [requested](https://github.com/brave/security/issues/new/choose) one
- [ ] There is a [ticket](https://github.com/brave/brave-browser/issues) for my issue
- [ ] Used Github [auto-closing keywords](https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue) in the PR description above
- [ ] Wrote a good [PR/commit description](https://google.github.io/eng-practices/review/developer/cl-descriptions.html)
- [ ] Squashed any review feedback or "fixup" commits before merge, so that history is a record of what happened in the repo, not your PR
- [ ] Added appropriate labels (`QA/Yes` or `QA/No`; `release-notes/include` or `release-notes/exclude`; `OS/...`) to the associated issue
- [ ] Checked the PR locally: `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests`, `npm run lint`, `npm run gn_check`, `npm run tslint`
- [ ] Ran `git rebase master` (if needed)

## Reviewer Checklist:

- [ ] A security review [is not needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or a link to one is included in the PR description
- [ ] New files have MPL-2.0 license header
- [ ] Adequate test coverage exists to prevent regressions
- [ ] Major classes, functions and non-trivial code blocks are well-commented
- [ ] Changes in component dependencies are properly reflected in `gn`
- [ ] Code follows the [style guide](https://chromium.googlesource.com/chromium/src/+/HEAD/styleguide/c++/c++.md)
- [ ] Test plan is specified in PR before merging

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on
- [ ] All relevant documentation has been updated, for instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protections
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Custom-Headers
  - [ ] https://github.com/brave/brave-browser/wiki/Web-Compatibility-Exceptions-in-Brave
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide
  - [ ] https://github.com/brave/brave-browser/wiki/P3A

### CODE REVIEW

**Comment by goodov** [comment]
x64:
https://ci.brave.com/job/brave-browser-build-windows-x64/4444/testReport/(root)/ScriptTriggerWidevinePermissionRequestBrowserTest/windows_x64___test_browser___SuggestPermissionIfWidevineDetected/?

File: browser/widevine/widevine_permission_request_browsertest.cc [hunk 1]
```diff
@@ -260,6 +261,11 @@ class ScriptTriggerWidevinePermissionRequestBrowserTest
 
 IN_PROC_BROWSER_TEST_F(ScriptTriggerWidevinePermissionRequestBrowserTest,
                        SuggestPermissionIfWidevineDetected) {
```
Comment (darkdh) [inline browser/widevine/widevine_permission_request_browsertest.cc:263]: Shouldn't we disable test case like this?
```cpp
#if BUILDFLAG(IS_WIN) && defined(ARCH_CPU_X86)
#define MAYBE_SuggestPermissionIfWidevineDetected DISABLED_SuggestPermissionIfWidevineDetected
#else
#define MAYBE_SuggestPermissionIfWidevineDetected SuggestPermissionIfWidevineDetected
#endif
IN_PROC_BROWSER_TEST_F(Maybe_ScriptTriggerWidevinePermissionRequestBrowserTest,
   MAYBE_SuggestPermissionIfWidevineDetected) {
...
```

Comment (simonhong) [reply browser/widevine/widevine_permission_request_browsertest.cc:263]: Yeah, right. Thanks for spotting!

**Comment by simonhong** [comment]
Updated to disable on Windows official build only.

File: browser/widevine/widevine_permission_request_browsertest.cc [hunk 2]
```diff
@@ -258,8 +258,16 @@ class ScriptTriggerWidevinePermissionRequestBrowserTest
...
 };
 
+#if defined(OFFICIAL_BUILD) && BUILDFLAG(IS_WIN)
+#define MAYBE_SuggestPermissionIfWidevineDetected \
+  DISABLED_SuggestPermissionIfWidevineDetected
+#else
+#define MAYBE_SuggestPermissionIfWidevineDetected \
+  SuggestPermissionIfWidevineDetected
+#endif
+
```
Comment (goodov) [inline browser/widevine/widevine_permission_request_browsertest.cc:268]: nit: usually there is no empty line between `MAYBE_` defines and the test itself.

**Review by goodov — APPROVED** [review]

File: browser/widevine/widevine_permission_request_browsertest.cc (hunk 2 above)
Comment (simonhong) [reply browser/widevine/widevine_permission_request_browsertest.cc]: Updated.

**Pull request merged**